packages/flutter/lib/src/material/text_form_field.dart

```diff
@@ -18,8 +18,8 @@ export 'package:flutter/services.dart' show SmartQuotesType, SmartDashesType;
 ///
 /// A [Form] ancestor is not required. The [Form] simply makes it easier to
 /// save, reset, or validate multiple fields at once. To use without a [Form],
-/// pass a [GlobalKey] to the constructor and use [GlobalKey.currentState] to
-/// save or reset the form field.
+/// pass a [GlobalKey<FormFieldState>] to the constructor and use
```

Comment:
This isn't going to generate the link you would like. If you want it to link to `GlobalKey` docs, it has to just be `[GlobalKey]`.

Probably you'll have to just add the link after the symbol, like below, or you can just skip the class link entirely.

```suggestion
/// pass a `GlobalKey<FormFieldState>` (see [GlobalKey]) to the constructor and use
```

Reply:
And my suggestions here also won't work? Apologies that I don't know the syntax and rules for this.

https://github.com/flutter/flutter/pull/105605#issuecomment-1150109153

Reply:
>  - `[GlobalKey]<FormFieldState>`
>  - `[GlobalKey]<[FormFieldState]>`

Well, they might look kind of odd, since the parts not in the link won't be in code font.  You can also run the docs generation and check it out locally: just run `dev/bots/docs.sh`, and the output will appear in `dev/docs/doc`.  Launch `index.html` there, and then navigate (you won't be able to search) to the class docs for this class.  My guess is that just leaving out the link will have the best results (i.e. just using `` `GlobalKey<FormFieldState>` ``). You might get away with something like this, but I haven't tried it:

```suggestion
/// pass a [GlobalKey]`<`[FormFieldState]`>` to the constructor and use
```

Reply:
Unfortunately, when I try that command, I consistently get an exception on [this line](https://github.com/flutter/flutter/blob/master/dev/bots/docs.sh#L34), even after completely wiping my `flutter/` directory and re-downloading (and reviewing my PATH variables). `flutter doctor` is fine. If that's just on my machine, I understand it's not in y'all's scope to do tech support. But maybe confirm that's working for you if you haven't already?

For now, without the ability to experiment, your initial suggestion seems like the best path.

Reply:
Yeah, `flutter` is in my PATH. It was in my user variables, but I added it to system variables, restarted, and got the same result. I can run `flutter` from anywhere and see it work.

I seem to hit these unusual bugs sometimes. My gut is that the user or context that `Process.run*()` does somehow isn't the same. I use Git Bash.

Reply:
Ahh, the problem is that the `docs.sh` script isn't designed to run on Windows (it needs to execute `flutter.bat`, and it's trying to execute `flutter`, a bash script, using the cmd shell). Sorry, I didn't notice you were on Windows.

It wouldn't be too hard to make it run on Windows, we just haven't (since it mainly runs on our CI infrastructure, which is Unix-based).  In any case, I think this PR is fine to land as-is.